release-23.1: backupccl: dont validate introduced tenant spans

msbutler opened the pull request:
Backport 1/1 commits from https://github.com/cockroachdb/cockroach/pull/108743 on behalf of @msbutler.

/cc https://github.com/orgs/cockroachdb/teams/release

---

In restore, we run a validation step in checkMissingIntroducedSpans() in which
we decode each introduced span in the backup manifests. This decoding step
naively assumed each introduced span was a table span. Incremental backups of
tenants, however, contain introduced tenant key spans, causing the validation
step to inadvertently fail during a restore of a tenant.

This patch skips this validation step for introduced tenant spans as it
does not apply to introduced tenant spans.

Epic: None

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/108625)
<!-- Reviewable:end -->

msbutler committed:
backupccl: dont validate introduced tenant spans

In restore, we run a validation step in checkMissingIntroducedSpans() in which
we decode each introduced span in the backup manifests. This decoding step
naively assumed each introduced span was a table span. Incremental backups of
tenants, however, contain introduced tenant key spans, causing the validation
step to inadvertently fail during a restore of a tenant.

This patch skips this validation step for introduced tenant spans as it
does not apply to introduced tenant spans.

Epic: None

Release note: none

msbutler commented on pkg/ccl/backupccl/targets.go:
```diff
@@ -428,6 +428,13 @@ func checkMissingIntroducedSpans(
+			} else if len(rest) == 0 {
+				// The key span represents a whole tenant's key space. Checking for
+				// introduced tables does not apply.
+				continue
```
@dt is this comment sufficient?
